From: http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/aa2a7493b202
# HG changeset patch
# User Laszlo Ersek <lersek@redhat.com>
# Date 1305548805 -3600
# Node ID aa2a7493b2027936d4b6224a324f8a7ae65a8e95
# Parent  c896d26c6b7c70bc82e353e7d02579fdd26af2f6
Subject: xenbus: fix xenbus_transaction_start() hang caused by double xenbus_transaction_end()
Patch-mainline: n/a

vbd_resize() up_read()'s xs_state.suspend_mutex twice in a row via
double xenbus_transaction_end() calls. The next down_read() in
xenbus_transaction_start() (at eg. the next resize attempt) hangs.

See RHBZ#618317.

Signed-off-by: Laszlo Ersek <lersek@redhat.com>
Acked-by: Jan Beulich <jbeulich@novell.com>
Acked-by: Ian Campbell <ian.campbell@citrix.com>

--- sle11sp1-2011-05-20.orig/drivers/xen/blkback/vbd.c	2011-05-20 12:34:30.000000000 +0200
+++ sle11sp1-2011-05-20/drivers/xen/blkback/vbd.c	2011-05-20 13:45:10.000000000 +0200
@@ -156,6 +156,7 @@ again:
 		goto again;
 	if (err)
 		printk(KERN_WARNING "Error ending transaction");
+	return;
 abort:
 	xenbus_transaction_end(xbt, 1);
 }
